packages/styles/src/elements/segmented-button.scss

```diff
@@ -0,0 +1,195 @@
+@use '../functions/tokens';
+@use '../tokens/components';
+@use '../mixins/utilities';
+@use '../core' as post;
+
+tokens.$default-map: components.$post-segmented-button;
+
+$post-segmented-button-max-count: 8;
+
+.segmented-button-container {
+  container-name: post-segmented-container;
+  container-type: inline-size;
+}
+
+legend {
+  @include post.visually-hidden();
+}
+
+.segmented-button {
+  display: flex;
+  gap: tokens.get('button-segmented-gap-inline');
+  align-items: stretch;
+  outline: tokens.get('button-segmented-border-width') solid
+    tokens.get('button-segmented-enabled-border');
+  outline-offset: calc(tokens.get('button-segmented-border-width') * -1);
+  background-color: tokens.get('button-segmented-enabled-bg');
+  border-radius: tokens.get('button-segmented-horizontal-border-radius');
+  box-shadow: tokens.get('button-segmented-elevation');
+
+  label {
+    flex: 0 1 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: relative;
+    z-index: 2;
+    padding-inline: tokens.get('button-segmented-padding-inline');
+    height: tokens.get('button-segmented-elements-height');
+    border-radius: inherit;
+    cursor: pointer;
+    font-weight: tokens.get('button-segmented-font-weight');
+    color: tokens.get('button-segmented-enabled-fg');
+    text-align: center;
+    line-height: 1.2;
+    overflow-wrap: anywhere;
+
+    input {
+      appearance: none !important;
+      user-select: none;
+      pointer-events: none;
+      position: absolute;
+      inset: 0;
+      margin: 0;
+      padding: 0;
+      border: tokens.get('button-segmented-border-width') solid transparent;
+      border-radius: inherit;
+      @include utilities.focus-style();
+    }
+
+    &:last-of-type {
+      z-index: 1;
+    }
+
+    &:not(:last-of-type) {
+      &:hover {
+        z-index: 3;
+      }
+    }
+
+    &:hover {
+      color: tokens.get('button-segmented-hover-fg');
+
+      input {
+        border-color: tokens.get('button-segmented-hover-border');
+      }
+
+      post-icon {
+        color: inherit;
+      }
```

Comment:
Post icon should already be inheriting color and I don't see any change if I disable this in the browser. Is this style really needed?